tools/clustering/cti-cluster-info.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+#
+# ----------------------------------------------------------------------------
+#
+# Copyright 2021 CHOPSTIX Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# ----------------------------------------------------------------------------
+#
+import argparse
+from src.clustering import ClusteringInformation
+
+def summary(args):
+    cluster = ClusteringInformation.from_file(args.cluster_file)
+
+    print("Cluster information parsed.")
+    print("Epsilon parameter: ", cluster.get_epsilon())
+    print("Invocation count: %d (in %d sets)" % (cluster.get_invocation_count(), cluster.get_invocation_set_count()))
+    print("Cluster count: ", cluster.get_cluster_count())
+    print("Noise invocations: %d (in %d sets)" % (cluster.get_noise_invocation_count(), cluster.get_noise_invocation_set_count()))
+
+def noise(args):
+    cluster_info = ClusteringInformation.from_file(args.cluster_file)
+
+
+def representative(args):
+    cluster_info = ClusteringInformation.from_file(args.cluster_file)
+
+    if args.noise:
+        for invocation in cluster_info.get_random_noise_invocations():
+            print(invocation)
+    elif args.cluster != None:
+        print(cluster_info.get_random_invocation_in_cluster(args.cluster))
+    else:
+        for index in range(cluster_info.get_cluster_count()):
+            print(cluster_info.get_random_invocation_in_cluster(index))
+
+        for invocation in cluster_info.get_random_noise_invocations():
+            print(invocation)
```

Comment:
Right know we are returning a random representative, which is OK since it is expected that all elements in a cluster behave similarly. However, there are to drawbacks:
- It complicates validation (2 successive runs on the same data will generate different outputs)
- It might introduce error when for _sparse_ clusters. Let's say that for whatever reason, the clustering algorithm comes up with a cluster definition where the data points within the cluster are quite sparse (note that they are not for the entire data set and that's why they are in the same cluster)

I propose to return always the _center_ of the cluster. We can add a flag to select the behavior we want.